tribuo-classification/client/src/main/java/com/r3/conclave/sample/client/Client.java

```diff
@@ -0,0 +1,133 @@
+            return;
+        }
+        String inputType = args[0];
+        String constraint = args[2];
+
+        //connect to host server
+        DataInputStream fromHost;
+        DataOutputStream toHost;
+        while (true) {
+            try {
+                System.out.println("Attempting to connect to localhost:9999");
+                Socket socket = new Socket();
+                socket.connect(new InetSocketAddress(InetAddress.getLoopbackAddress(), 9999), 5000);
+                fromHost = new DataInputStream(socket.getInputStream());
+                toHost = new DataOutputStream(socket.getOutputStream());
+                break;
+            } catch (Exception e) {
+                System.err.println("Retrying: " + e.getMessage());
+                Thread.sleep(2000);
+            }
+        }
+
+        //take inputs from user, populate the input data object with training, testing data and input type - train or evaluate
+        InputData inputData = new InputData(inputType);
+
+        if(TRAIN.equals(inputType)) {
+            //load the dataset to input type object using the csv loader
+            LabelFactory labelFactory = new LabelFactory();
+            CSVLoader csvLoader = new CSVLoader<>(labelFactory);
+
+            //input attributes/labels supplied via the input data file. Last column (in this case "class") is the classification output column name
+            String[] breastCancerHeaders = new String[]{"SampleCodeNumber", "ClumpThickness", "UniformityOfCellSize", "UniformityOfCellShape", "MarginalAdhesion",
+                    "SingleEpithelialCell Size", "BareNuclei", "BlandChromatin", "NormalNucleoli", "Mitoses", "Class"};
+
+            String fileName = System.getProperty("user.dir")+"/data/"+args[1];
+            ListDataSource irisesSource = csvLoader.loadDataSource(Paths.get(fileName),"Class",breastCancerHeaders);
+
+            //split the input loaded data into training and testing data
+            TrainTestSplitter irisSplitter = new TrainTestSplitter<>(irisesSource,0.7,1L);
+            MutableDataset trainingDataset = new MutableDataset<>(irisSplitter.getTrain());
+            MutableDataset testingDataset = new MutableDataset<>(irisSplitter.getTest());
+
+            inputData.setTestingDataset(testingDataset);
+            inputData.setTrainingDataset(trainingDataset);
+        }
+
+        //retrieve the attestation object from host immediately after connecting to host
+        byte[] attestationBytes = new byte[fromHost.readInt()];
+        fromHost.readFully(attestationBytes);
+
+        //convert byte[] received from host to EnclaveInstanceInfo object
+        EnclaveInstanceInfo instanceInfo = EnclaveInstanceInfo.deserialize(attestationBytes);
+
+//        //verify attestation received by enclave against the enclave code hash which we have
+        EnclaveConstraint.parse("S:"+ constraint +" PROD:1 SEC:INSECURE" ).check(instanceInfo);
```

Comment:
The enclave code [constraint](https://docs.conclave.net/writing-hello-world.html#constraints) is specified using `C:<MRENCLAVE>`, rather than `S:<MRSIGNER>`.